generator/internal/format_method_comments.cc

```diff
@@ -258,32 +266,60 @@ bool CheckMethodCommentSubstitutions() {
   return all_substitutions_used;
 }
 
-std::string FormatStartMethodComments() {
-  return R"""(  // clang-format off
-  ///
+std::string FormatStartMethodComments(bool is_method_deprecated) {
+  auto constexpr kMethodCommentsPrefix = R"""(  // clang-format off
+  ///)""";
+
+  auto constexpr kMethodCommentsSuffix = R"""(  ///
+  // clang-format on
+)""";
+
+  auto constexpr kCommentBody = R"""(
   /// @copybrief $method_name$
   ///
   /// Specifying the [`NoAwaitTag`] immediately returns the
   /// [`$longrunning_operation_type$`] that corresponds to the Long Running
   /// Operation that has been started. No polling for operation status occurs.
   ///
   /// [`NoAwaitTag`]: @ref google::cloud::NoAwaitTag
+)""";
+
+  std::string const deprecation_comment =
+      is_method_deprecated ?
+                           R"""( @deprecated This RPC is deprecated.
   ///
+  ///)"""
+                           : "";
+
+  return absl::StrCat(kMethodCommentsPrefix, deprecation_comment, kCommentBody,
+                      kMethodCommentsSuffix);
+}
+
+std::string FormatAwaitMethodComments(bool is_method_deprecated) {
+  auto constexpr kMethodCommentsPrefix = R"""(  // clang-format off
+  ///)""";
+
+  auto constexpr kMethodCommentsSuffix = R"""(  ///
   // clang-format on
 )""";
-}
 
-std::string FormatAwaitMethodComments() {
-  return R"""(  // clang-format off
-  ///
+  auto constexpr kCommentBody = R"""(
   /// @copybrief $method_name$
   ///
   /// This method accepts a `$longrunning_operation_type$` that corresponds
   /// to a previously started Long Running Operation (LRO) and polls the status
   /// of the LRO in the background.
-  ///
-  // clang-format on
 )""";
+
+  std::string const deprecation_comment =
```

Comment:
same nit about preferring `auto constexpr` over `std::string const`. I actually don't know why or if we prefer. It just seems more consistent.

Reply:
I tried `auto constexpr`, but it doesn't compile with the condition. Seems like it is compile time evaluation, so it requires `is_method_deprecated` be a `constexpr` as well.

Reply:
Ack, what you did with `kDeprecationComment` is what I had in mind.